client/checkout/woopay/email-input-iframe.js

```diff
@@ -75,11 +75,17 @@ export const handleWooPayEmailInput = async (
 
 	//Checks if customer has clicked the back button to prevent auto redirect
 	const searchParams = new URLSearchParams( window.location.search );
+	const isSkipWoopayCookieSet = /skip_woopay=1/.test( document.cookie );
```

Comment:
This regex is actually not sufficient to verify the exact cookie. I don't think this could be used maliciously, but a plugin could theoretically create a cookie like `not_skip_woopay=1` and it'll still skip WooPay.

It could also create a cookie like `skip_woopay=10` and that will also match the regex.

Alternatively, _we might later add a cookie_ that does something like `if_condition_is_true_skip_woopay=1` and this regex will match that.

There are several ways to fix this, e.g. modify the regex to be something like `/(;|\b)+skip_woopay=1;/` (not verified to work, think of this as more of an idea) to explicitly check the boundaries around the cookie.

Alternatively we could `document.cookie.split('; ')` (cookies should be split by `'; '`) and iterate through the cookies that way to check more explicitly like `if ( c.split('=')[0] === 'skip_woopay' && c.split('=')[1] === '1' ) { return true; // or break or whatever }`.

I took the liberty of adding tests to check for both of the "malicious" conditions in 65a296a4ee9d7b90d7f3b7c712ef6e8f0111b543 as explicit demonstrations 🙂

Reply:
Good call out! `(;|\b)+skip_woopay=1;` works but when the cookie is alone it doesn't because the browser doesn't put `;` in the end. I don't want to overcomplicate this, and to make all of this easier to read and make the logic more explicit, I have:

- Used the `split` approach to check the key/value more strictly, and;
- Moved the 'should skip WooPay' helper to `utils.js`. That makes more sense to me given that logic is not specific to the direct checkout feature, plus, we're able to reuse it in both places that need it, so far.

I've applied these changes in https://github.com/Automattic/woocommerce-payments/pull/8570/commits/a2d48a3df7787369c94792592f4c02b07d9692aa.